.github/workflows/ci.yml

```diff
@@ -0,0 +1,61 @@
+name: Tests and publishing
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  release:
+    types:
+    - created
+
+jobs:
+
+  tests:
+    name: Tests ${{ matrix.os }} - Node ${{ matrix.node }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+        node:
+          - 12.x
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Install Node.js
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node }}
+    - name: Compile
+      run: npm install && npm run compile
+    - name: Run tests on Linux with xvfb-run
+      run: xvfb-run -a npm test
+      if: runner.os == 'Linux'
+    - name: Run tests on Windows or macOS
+      run: npm test
+      if: runner.os != 'Linux'
+
+  publishRelease:
+    name: Publish Release
+    needs: [tests]
+    if: startsWith(github.ref, 'refs/tags/')
```

Comment:
So this will run if a tag is pushed?

Reply:
No, only when a github release is created, the [triggers](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) above don't include [`on.push.tags`](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore).